ibis/expr/types/temporal.py

```diff
@@ -436,10 +439,22 @@ class TimestampColumn(TemporalColumn, TimestampValue):
 class IntervalValue(Value):
     def to_unit(self, target_unit: str) -> IntervalValue:
         """Convert this interval to units of `target_unit`."""
-
         # TODO(kszucs): should use a separate operation for unit conversion
         # which we can rewrite/simplify to integer multiplication/division
-        return ops.ToIntervalUnit(self, unit=target_unit).to_expr()
```

Comment:
This operation wasn't implemented in any of the backends. The updated "rewrite" tries to be accurate, but the `else` branch the cast may not be well-defined.